src/Rules/Deprecations/OverrideDeprecatedPropertyRule.php

```diff
@@ -0,0 +1,60 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Rules\Deprecations;
+
+use PhpParser\Node;
+use PhpParser\Node\Stmt\Property;
+use PHPStan\Analyser\Scope;
+use PHPStan\Rules\Rule;
+use function sprintf;
+
+/**
+ * @implements Rule<Property>
+ */
+class OverrideDeprecatedPropertyRule implements Rule
+{
+
+	public function getNodeType(): string
+	{
+		return Property::class;
+	}
+
+	public function processNode(Node $node, Scope $scope): array
+	{
+		if (DeprecatedScopeHelper::isScopeDeprecated($scope)) {
+			return [];
+		}
+
+		if (!$scope->isInClass()) {
+			return [];
+		}
+
+		$class = $scope->getClassReflection();
+
+		$parents = $class->getParents();
+
+		$propertyName = (string) $node->props[0]->name;
```

Comment:
Is this the correct way to access the property name? Or is there a situation where there are more props in the array?

Reply:
Yeah, sure:

```php
class Foo
{
    
    private $foo, $bar = 1;
    
}
```

You should go through all the properties, not just the first one.